vuu/src/test/scala/org/finos/vuu/viewport/sessiontable/EditSessionTableTest.scala

```diff
@@ -0,0 +1,163 @@
+package org.finos.vuu.viewport.sessiontable
+
+import org.finos.toolbox.jmx.{MetricsProvider, MetricsProviderImpl}
+import org.finos.toolbox.time.{Clock, TestFriendlyClock}
+import org.finos.vuu.client.messages.RequestId
+import org.finos.vuu.core.table.TableTestHelper.{combineQs, emptyQueues}
+import org.finos.vuu.core.table.{RowWithData, TableContainer, ViewPortColumnCreator}
+import org.finos.vuu.net.ClientSessionId
+import org.finos.vuu.net.rpc.{EditRpcHandler, RpcHandler}
+import org.finos.vuu.util.table.TableAsserts.assertVpEq
+import org.finos.vuu.viewport.{AbstractViewPortTestCase, CloseDialogViewPortAction, DefaultRange, NoAction, OpenDialogViewPortAction, RenderComponent, SelectionViewPortMenuItem, ViewPort, ViewPortAction, ViewPortAddRowAction, ViewPortDeleteCellAction, ViewPortDeleteRowAction, ViewPortEditAction, ViewPortEditCellAction, ViewPortEditFailure, ViewPortEditRowAction, ViewPortEditSuccess, ViewPortFormCloseAction, ViewPortFormSubmitAction, ViewPortMenu, ViewPortSelectedIndices, ViewPortSelection, ViewPortTable}
+import org.scalatest.GivenWhenThen
+import org.scalatest.matchers.should.Matchers
+import org.scalatest.prop.Tables.Table
+
+import java.lang
+
+class EditSessionTableTest extends AbstractViewPortTestCase with Matchers with GivenWhenThen with AbstractSessionTestCase {
+
+  final implicit val clock: Clock = new TestFriendlyClock(TEST_TIME)
+  //implicit val metrics: MetricsProvider = new MetricsProviderImpl
+
+  /**
+   * This is the process rpc service.
+   */
+  class ProcessRpcService(tableContainer: TableContainer, clock: Clock) extends RpcHandler {
+
+    final val FIX_SEQ_RESET_TABLE = "fixSequenceReset"
+
+    private def openEditSeqNum(selection: ViewPortSelection, session: ClientSessionId): ViewPortAction = {
+
+      val baseTable = tableContainer.getTable(FIX_SEQ_RESET_TABLE)
+
+      val sessionTable = tableContainer.createSimpleSessionTable(baseTable, session)
+
+      val row = selection.rowKeyIndex.keys.map(selection.viewPort.table.pullRow(_)).toList.head
+
+      val processId = row.get("id").toString
+
+      sessionTable.processUpdate(processId, RowWithData(processId, Map("process-id" -> processId, "sequenceNumber" -> 0)), clock.now())
+
+      OpenDialogViewPortAction(ViewPortTable(sessionTable.name, sessionTable.tableDef.getModule().name), RenderComponent.InlineForm)
+    }
+
+    override def menuItems(): ViewPortMenu = ViewPortMenu("Admin",
+      new SelectionViewPortMenuItem("Reset SeqNum", "", this.openEditSeqNum, "OPEN_EDIT_RESET_FIX")
```

Comment:
this is defined as a Selection Menu Item. If user selects multiple rows before invoking action, will resetSequence be applied to all selected process Ids.
OR should this be a Row MenuItem ? Or is should we use Selection Menu Item and enforce single selection only in table ?

Reply:
At the moment we take the head row, I think we should have a category of single selection only yes. I guess we can add that tho. At the moment we could send back a validation error if more than one row.